java/src/main/java/com/saucelabs/simplesauce/SauceOptions.java

```diff
@@ -62,73 +50,56 @@ public SauceOptions withWindows7() {
     }
 
     public SauceOptions withEdge() {
-        browserName = "Edge";
-        browserVersion = "18.17763";
-        return this;
+        return withEdge18();
     }
 
-    //TODO notice the duplication below with edge.
-    //Maybe could be moved to a separate class so we can do withEdge().16_16299();
-    //Or withEdge().version(EdgeVersion.16_16299);
     public SauceOptions withEdge18() {
-        withEdge();
-        browserVersion = "18.17763";
-        return this;
+        return withEdge(EdgeVersion._18);
     }
     public SauceOptions withEdge17() {
-        withEdge();
-        browserVersion = "17.17134";
-        return this;
+        return withEdge(EdgeVersion._17);
     }
+
     public SauceOptions withEdge16() {
-        withEdge();
-        browserVersion = "16.16299";
-        return this;
+        return withEdge(EdgeVersion._16);
     }
 
     public SauceOptions withEdge15() {
-        withEdge();
-        browserVersion = "15.15063";
-        return this;
+        return withEdge(EdgeVersion._15);
     }
 
     public SauceOptions withEdge14() {
-        withEdge();
-        browserVersion = "14.14393";
-        return this;
+        return withEdge(EdgeVersion._14);
     }
 
     public SauceOptions withEdge13() {
-        withEdge();
-        browserVersion = "13.10586";
-        return this;
+        return withEdge(EdgeVersion._13);
     }
 
-    public SauceOptions withFirefox() {
-        browserName = "Firefox";
+    public SauceOptions withEdge(EdgeVersion version) {
+        browserName = "MicrosoftEdge";
+        browserVersion = version.getVersion();
         return this;
     }
 
-    public SauceOptions withIE(IEVersion version) {
-        this.browserVersion = version.getVersion();
+    public SauceOptions withFirefox() {
+        browserName = "firefox";
         return this;
     }
 
-    public SauceOptions withIE(String version) {
-        browserName = "IE";
-        browserVersion = version;
+    public SauceOptions withIE(IEVersion version) {
+        browserName = "internet explorer";
```

Comment:
Wow, will this work as it used to be "IE"?

Reply:
Honestly, I don't know how "ie" works, I don't think it should.